### DISCUSSION TITLE
Removing the `search/completion` handler

### DISCUSSION THREAD

**Discussion by 4e6** [enso-org/enso#5662] [Work In Progress - GUI] [answered]
I will remove the [`search/completion`](https://github.com/enso-org/enso/blob/f316ba01367dd7bf25b63f0bd4f774fb105e3d13/docs/language-server/protocol-language-server.md#searchcompletion) handler from the language server in my upcoming PR. It will simplify the logic and should positively impact the startup time.
As far as I can see, while it is still used in IDE, it is superseded by the new component browser.

Here is the related issue to cleanup the IDE codebase #5661

**Comment by farmaazon** [comment, marked as answer]
> I will remove the [`search/completion`](https://github.com/enso-org/enso/blob/f316ba01367dd7bf25b63f0bd4f774fb105e3d13/docs/language-server/protocol-language-server.md#searchcompletion) handler from the language server in my upcoming PR. It will simplify the logic and should positively impact the startup time. As far as I can see, while it is still used in IDE, it is superseded by the new component browser.
> 
> Here is the related issue to cleanup the IDE codebase #5661

@4e6 no, the search/completion **is not** superseded yet. We have a task for doing so, but it's not ready yet.

**Reply by 4e6** [reply]
Thanks for the heads up. For now, created an appropriate task for the engine #5666 instead

**Reply by farmaazon** [reply]
The task you linked mentions that 

> The task should be implemented after IDE drops its support in https://github.com/enso-org/enso/issues/5661

5661 _will not_ remove all of our usages of search/completion, as the new Component Browser also uses it. 

> We have a task for doing so, but it's not ready yet.

I checked, and actually the task is not yet created. We thought about that, but implementing it will require actually a "database-like" implementation, and we wanted to check if we can use some existing implementation (also check if those implementations are quick).

**Reply by 4e6** [reply]
Ugh, this sounds more complicated than I thought. If we're not sure when (if) this "database-like" storage is implemented, I'll close the engine task for now. Because when it's implemented, we'll have a better idea of what should be done on the engine side to support it.

**Reply by farmaazon** [reply]
> Ugh, this sounds more complicated than I thought. If we're not sure when (if) this "database-like" storage is implemented, I'll close the engine task for now. Because when it's implemented, we'll have a better idea of what should be done on the engine side to support it.

Actually, we think about storage on the IDE side only, the ls API would stay the same.
